helpers/integration.go

```diff
@@ -236,6 +236,8 @@ func CalculateIntegrationPipelineRunOutcome(adapterClient client.Client, ctx con
 		if err != nil {
 			return false, fmt.Errorf("error while getting test results from pipelineRun %s: %w", pipelineRun.Name, err)
 		}
+	} else if HasPipelineRunFinished(pipelineRun) && !HasPipelineRunSucceeded(pipelineRun) {
+		return false, fmt.Errorf("PipelineRun %s in namespace %s failed for %s", pipelineRun.Name, pipelineRun.Namespace, GetPipelineRunFailedReason(pipelineRun))
```

Comment:
I think that if we return an error here, it will break the loop [here](https://github.com/redhat-appstudio/integration-service/blob/e86f164b6828d7d29bfd87378f1f8ff74ed83463/controllers/pipeline/pipeline_adapter.go#L391-L394) and ultimately get picked up by [this check](https://github.com/redhat-appstudio/integration-service/blob/e86f164b6828d7d29bfd87378f1f8ff74ed83463/controllers/pipeline/pipeline_adapter.go#L165-L169), ending the Reconciliation loop and not updating the Snapshot status as a result.

Maybe we can log the output of `GetPipelineRunFailedReason(pipelineRun)` before returning `false, nil` here instead?

Reply:
Updated. Thanks!